app/src/main/java/com/mithridat/nonoconverter/backend/nonogram/Field.java

```diff
@@ -0,0 +1,339 @@
+     */
+    public int getLength(int type) {
+        if (type == ROW) return getRows();
+        return getCols();
+    }
+
+    /**
+     * Method for getting field cell color with coordinates (i,j)
+     *
+     * @param i - field cell row number
+     * @param j - field cell column number
+     * @return field cell color with coordinates (i,j)
+     */
+    public int getColor(int i, int j) {
+        return _field[i][j];
+    }
+
+    /**
+     * Method for getting field cell color with coordinates (i,j) or (j, i)
+     *
+     * @param i - i
+     * @param j - j
+     * @param type - ROW, if (i, j)
+     *               COL, if (j, i)
+     * @return field cell color with coordinates (i,j) or (j, i)
+     */
+    public int getColor(int i, int j, int type) {
+        if (type == ROW) return getColor(i, j);
+        return getColor(j, i);
+    }
+
+    /**
+     * Method for getting field cell color with coordinates (i,j) or (j, i)
+     *
+     * @param i - i
+     * @param j - j
+     * @param type - ROW, if (i, j)
+     *               COL, if (j, i)
+     * @param state - state
+     * @return field cell color with coordinates (i,j) or (j, i)
+     */
+    public int getColorState(int i, int j, int type, int state) {
+        int color = getColor(i, j, type);
+        if (color == EMPTY) return EMPTY;
+        if (state == _state) return color;
+        if (_state == INDEX) return _colors[color];
+        return Arrays.binarySearch(_colors, color);
+    }
+
+    /**
+     * Method for setting field cell color with coordinates (i,j)
+     *
+     * @param i - field cell row number
+     * @param j - field cell column number
+     * @param color - color
+     */
+    public void setColor(int i, int j, int color) {
+        _field[i][j] = color;
+    }
+
+    /**
+     * Method for setting field cell color with coordinates (i,j) or (j, i)
+     *
+     * @param i - i
+     * @param j - j
+     * @param color - color
+     * @param type - ROW, if (i, j)
+     *               COL, if (j, i)
+     */
+    public void setColor(int i, int j, int color, int type) {
+        if (type == ROW) setColor(i, j, color);
+        setColor(j, i, color);
+    }
+
+    /**
+     * Method for getting count of colors
+     *
+     * @return count of colors
+     */
+    public int getColorsCount() {
+        return _colorsCount;
+    }
+
+    /**
+     * Method for getting index of first cell with another color
+     * (current color - the color of cell with index = pos) in row or column
+     *
+     * @param ind - index of row or column
+     * @param pos - position, the search begins with the next cell
+     * @param dir - direction, +1 or -1
+     * @param type - ROW, if row
+     *               COL, if column
+     * @return index of first cell with another color in row or column
+     */
+    public int getAnotherColorIndex(int ind, int pos, int dir, int type) {
+        int i = pos + 1, len = getLength(ROW + COL - type);
```

Comment:
Why `ROW + COL - type`?

Reply:
Because it was: `len = _cols` if `type == ROW` and `len = _rows` if `type == COL`.

Reply:
Ok. Is it possible to change the semantic of the method `getLength` or create a new method? New semantic: if the type is `ROW` method returns a count of columns i.e. length of a row.